tcms/testruns/tests/test_admin.py

```diff
@@ -22,3 +28,20 @@ def test_regular_user_can_not_change_testrun(self):
         response = self.client.get(reverse('admin:testruns_testrun_change',
                                            args=[self.test_run.pk]))
         self.assertRedirects(response, reverse('testruns-get', args=[self.test_run.pk]))
+
+    def test_regular_user_can_not_delete_testrun(self):
+        response = self.client.get(reverse('admin:testruns_testrun_delete',
+                                           args=[self.test_run.pk]))
+        self.assertRaisesMessage(response, ('Permission denied: TestRun does not belong to you'))
+
+    def test_superuser_can_delete_testrun(self):
+        self.client.login(
+            username=self.superuser.username,
+            password='password')
+        response = self.client.get(reverse('admin:testruns_testrun_delete',
+                                           args=[self.test_run.pk]))
+        self.assertContains(response, ("Yes, I'm sure"))
+
+        response = self.client.post(reverse('admin:testruns_testrun_delete',
+                                            args=[self.test_run.pk]), {'post': 'yes'})
+        self.assertRedirects(response, reverse('admin:testruns_testrun_changelist'))
```

Comment:
This test method is generally OK but needs additional work:

- After deletion you need to verify that TR has actually been removed from the DB
- ATM I'm not sure if removing the test object introduces dependencies between the test methods and the order in which they are executed. I'll have to consult the docs to see how the transaction engine handles the deletion. The easiest way for you to check that is to have another test method which will execute after this one and will assert that this TR still exists in the DB. Django test runner sorts methods by name so something like `test_zzzz` should be executed after the current method.